mongorestore/restore.go

```diff
@@ -69,6 +69,150 @@ func NewResultFromBulkResult(result *mongo.BulkWriteResult, err error) Result {
 	return Result{nSuccess, nFailure, err}
 }
 
+func (restore *MongoRestore) RestoreIndexesForIntents() error {
```

Comment:
are there any key things to review in this file? It looks like a lot of it is refactoring, but it's hard to tell from the diff

Reply:
Not sure I can narrow it down. I'd love some eyes on the refactoring itself to double check it's correct. Previously reading metadata files, copying data, and index builds all happened in `RestoreIntent()`. Now we cycle through the intents 3 times and do each task separately. Copying data and index builds are parallelized but reading metadata files is single threaded.